src/py/flwr/common/record/parametersrecord.py

```diff
@@ -268,103 +280,288 @@ def _check_value(value: Array) -> None:
-    persistent state of a :code:`ClientApp` via its context. Regardless of the usecase,
-    we will sooner or later want to recover the array in its original NumPy
-    representation. For the example above, where the array was serialized using the
-    built-in utility function, deserialization can be done as follows:
-
-    >>> # Use the Array's built-in method
-    >>> arr_np_d = arr.numpy()
-    >>>
-    >>> # If printed, it will show the exact same data as above:
-    >>> array([[-1.84242409, -1.01539537, -0.46528405],
-    >>>      [ 0.32991896,  0.55540414,  0.44085534],
-    >>>      [-0.10758364,  1.97619858, -0.37120501]])
-
-    If you need finer control on how your arrays are serialized and deserialized, you
-    can construct :code:`Array` objects directly like this:
-
-    >>> from flwr.common import Array
-    >>> # Serialize your array and construct Array object
-    >>> arr = Array(
-    >>>         data=ndarray.tobytes(),
-    >>>         dtype=str(ndarray.dtype),
-    >>>         stype="",  # Could be used in a deserialization function
-    >>>         shape=list(ndarray.shape),
-    >>>       )
-    >>>
-    >>> # Then you can deserialize it like this
-    >>> arr_np_d = np.frombuffer(
-    >>>             buffer=array.data,
-    >>>             dtype=array.dtype,
-    >>>            ).reshape(array.shape)
-
-    Note that different arrays (e.g. from PyTorch, Tensorflow) might require different
-    serialization mechanism. Howerver, they often support a conversion to NumPy,
-    therefore allowing to use the same or similar steps as in the example above.
+    >>> arr1 = np.random.randn(3, 3)
+    >>> arr2 = np.random.randn(2, 2)
+    >>> p_record = ParametersRecord([arr1, arr2])
+
+    Initializing with a PyTorch model state_dict:
+
+    >>> import torch.nn as nn
+    >>> model = nn.Linear(10, 5)
+    >>> p_record = ParametersRecord(model.state_dict())
+
+    Initializing with a TensorFlow model weights (a list of NumPy arrays):
+
+    >>> import tensorflow as tf
+    >>> model = tf.keras.Sequential([tf.keras.layers.Dense(5, input_shape=(10,))])
+    >>> p_record = ParametersRecord(model.get_weights())
     """
 
-    def __init__(
+    @overload
+    def __init__(self) -> None: ...  # noqa: E704
+
+    @overload
+    def __init__(  # noqa: E704
+        self, array_dict: OrderedDict[str, Array], *, keep_input: bool = True
+    ) -> None: ...
+
+    @overload
+    def __init__(  # noqa: E704
+        self, numpy_ndarrays: list[NDArray], *, keep_input: bool = True
+    ) -> None: ...
+
+    @overload
+    def __init__(  # noqa: E704
+        self, state_dict: OrderedDict[str, torch.Tensor], *, keep_input: bool = True
+    ) -> None: ...
+
+    def __init__(  # pylint: disable=too-many-arguments
         self,
+        *args: Any,
+        numpy_ndarrays: list[NDArray] | None = None,
+        state_dict: OrderedDict[str, torch.Tensor] | None = None,
```

Comment:
These two have inconsistent naming, two consistent options would be:
- `numpy_ndarrays` & `torch_state_dict`
- `ndarrays` & `state_dict`